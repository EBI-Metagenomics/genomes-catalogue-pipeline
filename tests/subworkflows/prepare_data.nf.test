nextflow_workflow {

    name "Test Workflow PREPARE_DATA"
    script "subworkflows/prepare_data.nf"
    workflow "PREPARE_DATA"

    test("Should filter and rename genomes") {

        when {
            params {
                input_genomes = "$baseDir/tests/modules/fixtures/classify_clusters/dereplicated_genomes"
                input_csv = "$baseDir/tests/modules/fixtures/filter_qs50/genomes.csv"
            }
            workflow {
                """
                input[0] = channel.fromPath("${params.input_genomes}", checkIfExists: true)
                input[1] = channel.fromPath("${params.input_csv}", checkIfExists: true)
                input[2] = channel.empty()
                input[3] = channel.value(1)
                input[4] = channel.value(8)
                input[5] = channel.empty()
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Should use the genome_info file") {
        when {
            params {
                input_genomes = "$baseDir/tests/modules/fixtures/classify_clusters/dereplicated_genomes"
                input_csv = "$baseDir/tests/modules/fixtures/filter_qs50/genomes.csv"
                per_genome_category = "$baseDir/tests/subworkflows/fixtures/prepare_data/"
            }
            workflow {
                """
                input[0] = channel.fromPath("${params.input_genomes}", checkIfExists: true)
                input[1] = channel.fromPath("${params.input_csv}", checkIfExists: true)
                input[2] = channel.empty()
                input[3] = channel.value(1)
                input[4] = channel.value(8)
                input[5] = channel.empty()
                input[6] = channel.fromPath("${params.per_genome_category}", checkIfExists: true)
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }
}
