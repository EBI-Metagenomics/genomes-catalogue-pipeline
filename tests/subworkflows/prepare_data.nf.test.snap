{
    "Should filter and rename genomes": {
        "content": [
            {
                "genomes_checkm": [

                ],
                "genomes": [

                ],
                "genomes_name_mapping": [

                ]
            }
        ],
        "timestamp": "2023-04-25T13:43:42+0000"
    }
}
